bin/cron/pardot_sync

```diff
@@ -0,0 +1,419 @@
+    street_address = street_address[0...255] unless street_address.nil?
+    city = city[0...255] unless city.nil?
+    state = state[0...255] unless state.nil?
+    postal_code = postal_code[0...255] unless postal_code.nil?
+    country = country[0...255] unless country.nil?
+
+    # add to batch to update
+    update_batch += conn[:contacts_rollup].where(:email => email).update_sql(
+      :street_address => street_address, :city => city, :state => state, :postal_code => postal_code, :country => country) + ";"
+
+    num_updated += 1
+
+    if record_count % UPDATE_BATCH_SIZE == 0
+      begin
+        conn.run update_batch
+      rescue => e
+        log "Error: #{e}. Cmd: #{update_batch}"
+        raise
+      end
+
+      # Discard the connection and get a new one after each batch. Currently, if you use multiple statements in a batch, the first call will succeed; the next call on
+      # the same connecton AFTER you do a multi-stament call will fail with a MySQL error that commands are out of order. The innertubes suggest that this may be a problem in mysql2 gem.
+      # Fortunately, it is fairly cheap to get a new connection for each batch (time to get new connection is negligible compared to batch time).
+      conn.disconnect
+      conn = mysql_multi_connection
+
+      update_batch = ""
+      if Time.now - time_last_output > 5
+        log "#{record_count} "
+        time_last_output = Time.now
+      end
+    end
+  end
+
+  conn.run update_batch unless update_batch.empty?
+  log "- #{record_count} source records, #{num_updated} with geo data, #{num_parse_errors} skipped due to invalid JSON, " + ((Time.now) - start).to_s + " seconds"
+end
+
+def update_grades_taught
+  start = Time.now
+  log "Updating grades taught from dashboard.users"
+  PEGASUS_DB.run "
+    UPDATE contacts_rollup,
+    (SELECT dashboard_user_id, group_concat(grade) as grades FROM
+	    (SELECT DISTINCT contacts_rollup.dashboard_user_id, sections.grade from contacts_rollup
+		    INNER JOIN #{DASHBOARD_DB_NAME}.sections AS sections on sections.user_id = contacts_rollup.dashboard_user_id
+		    WHERE contacts_rollup.dashboard_user_id is not null
+		    AND length(sections.grade) > 0
+		    ORDER by 1,2
+	    ) q
+	    GROUP BY dashboard_user_id
+    ) src
+    SET contacts_rollup.grades_taught = src.grades
+    WHERE contacts_rollup.id > 0 AND (contacts_rollup.dashboard_user_id = src.dashboard_user_id)"
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def update_ages_taught_for_age(age)
+  start = Time.now
+  log "Updating ages taught for age #{age}"
+
+  min_birthday = ""
+  max_birthday = ""
+  # for age = 21, include 21 and above
+  min_birthday = "students.birthday > DATE_ADD(NOW(),INTERVAL -#{age + 1} YEAR) AND" unless age >= 21
+  # for age = 4, include 4 and below
+  max_birthday = "students.birthday <= DATE_ADD(NOW(),INTERVAL -#{age} YEAR) AND" unless age <= 4
```

Comment:
nit: Consider naming these `{min,max}_birthday_clause` or the like. As is, these sound like numbers.